src/main/java/ua/mate/cinema/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,71 @@
+package ua.mate.cinema.dao.impl;
+
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import ua.mate.cinema.dao.interfaces.ShoppingCartDao;
+import ua.mate.cinema.exception.DataProcessingException;
+import ua.mate.cinema.lib.Dao;
+import ua.mate.cinema.model.ShoppingCart;
+import ua.mate.cinema.model.User;
+import ua.mate.cinema.util.HibernateUtil;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession().getSession();
+            transaction = session.beginTransaction();
+            session.saveOrUpdate(shoppingCart);
+            transaction.commit();
+            return shoppingCart;
+        } catch (Exception e) {
+            if (transaction != null && transaction.isActive()) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add shopping cart " + shoppingCart, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession().getSession()) {
+            Query<ShoppingCart> getShoppingCartByUserQuery
+                    = session.createQuery("from ShoppingCart sc "
+                    + "left join fetch sc.tickets "
+                    + "where sc.user = :user", ShoppingCart.class);
+            getShoppingCartByUserQuery.setParameter("user", user);
+            return getShoppingCartByUserQuery.getSingleResult();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get shopping cart by user " + user, e);
+        }
```

Comment:
No need to catch an exception here
If we've got the user registered, the shopping cart definitely exists